yarn-project/prover-client/src/orchestrator/orchestrator.ts

```diff
@@ -770,26 +859,34 @@ export class ProvingOrchestrator {
     if (publicFunction.vmRequest) {
       // This function tries to do AVM proving. If there is a failure, it fakes the proof unless AVM_PROVING_STRICT is defined.
       // Nothing downstream depends on the AVM proof yet. So having this mode lets us incrementally build the AVM circuit.
-      const doAvmProving = async (signal: AbortSignal) => {
-        const inputs: AvmCircuitInputs = new AvmCircuitInputs(
-          publicFunction.vmRequest!.functionName,
-          publicFunction.vmRequest!.bytecode,
-          publicFunction.vmRequest!.calldata,
-          publicFunction.vmRequest!.kernelRequest.inputs.publicCall.callStackItem.publicInputs,
-          publicFunction.vmRequest!.avmHints,
-        );
-        try {
-          return await this.prover.getAvmProof(inputs, signal);
-        } catch (err) {
-          if (process.env.AVM_PROVING_STRICT) {
-            throw err;
-          } else {
-            logger.warn(`Error thrown when proving AVM circuit: ${err}`);
-            logger.warn(`AVM_PROVING_STRICT is off, faking AVM proof and carrying on...`);
-            return { proof: makeEmptyProof(), verificationKey: VerificationKeyData.makeFake() };
+      const doAvmProving = wrapCallbackInSpan(
+        this.tracer,
+        'ProvingOrchestrator.prover.getAvmProof',
+        {
+          [Attributes.TX_HASH]: txProvingState.processedTx.hash.toString(),
+          [Attributes.APP_CIRCUIT_NAME]: publicFunction.vmRequest!.functionName,
```

Comment:
Should we also log the artifact classId if we intend to group by function?

Reply:
This should be a `<contract_name>:<function_name>` label :D